c2h/include/c2h/custom_type.h

```diff
@@ -178,34 +179,33 @@ class accumulateable_t
 
 } // namespace c2h
 
-namespace std
-{
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
 template <template <typename> class... Policies>
-class numeric_limits<c2h::custom_type_t<Policies...>>
+class __numeric_limits_impl<c2h::custom_type_t<Policies...>, __numeric_limits_type::__other>
```

Comment:
Shouldnt this also work if we just specialized `numeric_limits`

Reply:
It doesn't and I need your guidance here. The problem is that CUB sometimes queries `numeric_limits<const T>` and that does not find a specialization if I only specialize:
```
class numeric_limits<c2h::custom_type_t<Policies...>> { ... };
```
Specializing `__numeric_limits_impl` works, because `numeric_limits` strips CV qualifiers and passes the type to `__numeric_limits_impl`. What's the best practice? Should be specialize `numeric_limits` multiple times for `T`, `const T`, and maybe also for referneces of those? Or should we try harder inside CUB to call like `numeric_limits<decay_t<T>>`?

Reply:
That is a bug in our implementation then: https://eel.is/c++draft/numeric.limits#general-5